src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoReportDataDaoTest.java

```diff
@@ -0,0 +1,248 @@
+import static org.sagebionetworks.bridge.models.reports.ReportType.PARTICIPANT;
+import static org.sagebionetworks.bridge.models.reports.ReportType.STUDY;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertSame;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.datamodeling.PaginatedQueryList;
+import com.amazonaws.services.dynamodbv2.datamodeling.QueryResultPage;
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.ImmutableList;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.LocalDate;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import org.sagebionetworks.bridge.models.DateRangeResourceList;
+import org.sagebionetworks.bridge.models.ForwardCursorPagedResourceList;
+import org.sagebionetworks.bridge.models.ResourceList;
+import org.sagebionetworks.bridge.models.reports.ReportData;
+import org.sagebionetworks.bridge.models.reports.ReportDataKey;
+
+public class DynamoReportDataDaoTest extends Mockito {
+
+    static final LocalDate START_DATE = LocalDate.parse("2016-03-28");
+    static final LocalDate END_DATE = LocalDate.parse("2016-03-31");
+    static final DateTime START_TIME = DateTime.parse("2016-03-28T17:16:28.711-07:00");
+    static final DateTime END_TIME = DateTime.parse("2016-03-31T17:16:28.711-07:00");
+    static final String REPORT_ID = "aReportId";
+    static final String OFFSET_KEY = "anOffsetKey";
+    static final ReportDataKey STUDY_REPORT_KEY = new ReportDataKey.Builder().withIdentifier(REPORT_ID)
+            .withReportType(STUDY).withStudyIdentifier(TEST_STUDY).build();
+    static final ReportDataKey PARTICIPANT_REPORT_KEY = new ReportDataKey.Builder().withIdentifier(REPORT_ID)
+            .withHealthCode(HEALTH_CODE).withReportType(PARTICIPANT).withStudyIdentifier(TEST_STUDY).build();
+    
+    static final DynamoReportData REPORT0 = createReport(DateTime.parse("2016-03-28T17:16:28.711-07:00"), "g", "h");
+    static final DynamoReportData REPORT1 = createReport(DateTime.parse("2016-03-29T17:16:28.711-07:00"), "a", "b");
+    static final DynamoReportData REPORT2 = createReport(DateTime.parse("2016-03-30T17:16:28.711-07:00"), "c", "d");
+    static final DynamoReportData REPORT3 = createReport(DateTime.parse("2016-03-31T17:16:28.711-07:00"), "e", "f");
```

Comment:
We should avoid using static constants that are expected to be modified as part of a test. This can cause one test to bleed into the next.

It would be better to make them non-static and initialize them in the BeforeMethod.

Reply:
changed to instance fields